Error while checking build/O2/fullCI for 6e8aa83dd5cad60f92fe49c0658f7a2a58292a4e at 2024-06-24 23:44:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/SOURCES/O2/13244-slc8_x86-64/0/Detectors/ITSMFT/common/reconstruction/src/AlpideCoder.cxx:129:20: error: request for member 'begin' in 'hits', which is of non-class type 'short int [1024]'
/sw/SOURCES/O2/13244-slc8_x86-64/0/Detectors/ITSMFT/common/reconstruction/src/AlpideCoder.cxx:129:34: error: request for member 'begin' in 'hits', which is of non-class type 'short int [1024]'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13244/6e8aa83dd5cad60f92fe49c0658f7a2a58292a4e/build_O2_fullCI/pretty.html).

--- a/Detectors/ITSMFT/common/reconstruction/src/AlpideCoder.cxx
+++ b/Detectors/ITSMFT/common/reconstruction/src/AlpideCoder.cxx
@@ -76,7 +76,7 @@ int AlpideCoder::encodeChip(PayLoadCont& buffer, const o2::itsmft::ChipPixelData
 int AlpideCoder::procDoubleCol(PayLoadCont& buffer, short reg, short dcol)
 {
   // process double column: encoding
-  short hits[2 * NRows];
+  std::array<short, 2 * NRows> hits;
   int nHits = 0, nData = 0;
   //
   int nr = mFirstInRow.size();
@@ -84,8 +84,11 @@ int AlpideCoder::procDoubleCol(PayLoadCont& buffer, short reg, short dcol)
   int prevRow = -1;
   for (int ir = 0; ir < nr; ir++) {
     int linkID = mFirstInRow[ir];
-    if (linkID == -1 || mPix2Encode[linkID].col > col1) { // no pixels left on this row or higher column IDs
+    if (linkID == -1) { // no pixels left on this row
       continue;
+    }
+    if (mPix2Encode[linkID].col > col1) { // all following hits will have higher columns
+      break;
     }
     short rowID = mPix2Encode[linkID].row;
     // process fired pixels
@@ -122,6 +125,9 @@ int AlpideCoder::procDoubleCol(PayLoadCont& buffer, short reg, short dcol)
   }
   //
   int ih = 0;
+  if (nHits > 1) {
+    std::sort(hits.begin(), hits.begin() + nHits);
+  }
   while ((ih < nHits)) {
     short addrE, addrW = hits[ih++]; // address of the reference hit
     uint8_t mask = 0;

